dev/com.ibm.jbatch.container/src/com/ibm/jbatch/container/services/impl/JPAPersistenceManagerImpl.java

```diff
@@ -2685,34 +2698,169 @@ private int getJobInstanceTableVersion(PersistenceServiceUnit psu) throws Except
             Integer exec = new TranRequest<Integer>(em) {
                 @Override
                 public Integer call() throws Exception {
-                    // Verify that UPDATETIME column exists by running a query against it.
-                    String queryString = "SELECT COUNT(x.lastUpdatedTime) FROM JobInstanceEntityV2 x";
+                    // Verify that groupNames column exists by running a query against it.
+                    String queryString = "SELECT COUNT(x.groupName) FROM JobInstanceEntityV3 x";
                     TypedQuery<Long> query = em.createQuery(queryString, Long.class);
                     query.getSingleResult();
-                    logger.fine("The UPDATETIME column exists, job instance table version = 2");
-                    instanceVersion = 2;
+                    logger.fine("The groupNames column exists, job instance table version = 3");
+                    instanceVersion = 3;
                     return instanceVersion;
                 }
```

Comment:
We should check that the V2 change is there as well before declaring this V3, otherwise we should fall back to V1.  Maybe you could turn this "inside out" , so to speak?

Reply:
check v2 first ... if v2 go on and check v3. if not v2 then set to v1